test/unit_tests/cvode/gtest/CMakeLists.txt

```diff
@@ -12,9 +12,36 @@
 
 # include location of public and private header files
 
-add_executable(test_error_handling test_error_handling.cpp)
-target_include_directories(test_error_handling PRIVATE
-  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include> ${CMAKE_SOURCE_DIR}/include ${CMAKE_SOURCE_DIR}/src)
-target_link_libraries(test_error_handling
-    PRIVATE  sundials_cvode sundials_nvecserial sundials_core GTest::gtest_main GTest::gmock)
-gtest_discover_tests(test_error_handling)
+add_executable(test_cvode_error_handling test_cvode_error_handling.cpp)
+target_include_directories(test_cvode_error_handling
+  PRIVATE
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include> 
+  ${CMAKE_SOURCE_DIR}/include 
+  ${CMAKE_SOURCE_DIR}/src
+)
+
+# We explicitly choose which object libraries to link to and link in the
+# cvode objects so that we have access to private functions w/o changing
+# their visibility in the installed libraries.
```

Comment:
Is this needed in ARKODE as well? I don't remember all these objects in that case

Reply:
It is needed there, I have made the change along with the other changes you noticed.